llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7777,6 +7565,18 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
 
       InstructionCost Cost = cost(*P, VF);
       VectorizationFactor CurrentFactor(VF, Cost, ScalarCost);
+
+      // Make sure that the VF doesn't use more than the number of available
+      // registers
+      const auto &MLU = RU.MaxLocalUsers;
+      if (any_of(MLU, [&](decltype(MLU.front()) &LU) {
```

Comment:
```suggestion
      if (any_of(MLU, [&](auto &LU) {
```
simpler to just use `auto`?

Reply:
Done.